Marlin/src/core/macros.h

```diff
@@ -317,6 +317,10 @@
 
   #endif
 
+  // C++11 solution that is standard compliant. type_traits is not available on all platform
+  template<bool, typename _Tp = void> struct ENABLE_IF { };
+  template<typename _Tp>              struct ENABLE_IF<true, _Tp> { typedef _Tp type; };
```

Comment:
I think the choice of the macro name IF_ENABLED is innappropriate. `enable_if` is a classic tool in std namespace so C++ developers might be confused by the IF_ENABLED macro. Anyway, I'll put the definition in a Private namespace so it'll not collide.

Reply:
We also have these in `types.h`:
```cpp
//
// Conditional type assignment magic. For example...
//
// typename IF<(MYOPT==12), int, float>::type myvar;
//
template <bool, class L, class R>
struct IF { typedef R type; };
template <class L, class R>
struct IF<true, L, R> { typedef L type; };
```

Reply:
It's not exactly the same. The version in my code is used for SFINAE (substitution failure is not an error) and it's not L or R depending on a condition. It's valid L or invalid code depending on the condition.

So that the invalid code is no more considered by the compiler when considering overloads, which fallback to another "catch-all" overload in that case that does nothing more than returning a configurable default value. 

In all cases, I've moved the declaration/definition in a private namespace so it does not conflict with existing code.